config/crd/bases/packages.glasskube.dev_packages.yaml

```diff
@@ -64,6 +64,59 @@ spec:
                 - name
                 - version
                 type: object
+              values:
+                additionalProperties:
+                  properties:
+                    value:
+                      type: string
+                    valueFrom:
+                      properties:
+                        configMapRef:
+                          properties:
+                            key:
+                              type: string
+                            name:
+                              type: string
+                            namespace:
+                              type: string
+                          required:
+                          - key
+                          - name
+                          type: object
+                        packageRef:
+                          properties:
+                            name:
+                              type: string
+                            value:
+                              type: string
+                          required:
+                          - name
+                          - value
+                          type: object
+                        secretRef:
+                          properties:
+                            key:
+                              type: string
+                            name:
+                              type: string
+                            namespace:
+                              type: string
+                          required:
+                          - key
+                          - name
+                          type: object
+                      type: object
+                      x-kubernetes-validations:
+                      - message: ValueReference must have exactly one of configMapRef,
+                          secretRef, packageRef
+                        rule: '[has(self.configMapRef), has(self.secretRef), has(self.packageRef)].exists_one(x,
+                          x)'
+                  type: object
+                  x-kubernetes-validations:
+                  - message: ValueConfiguration must have exactly one of value, valueFrom,
+                      packageRef
+                    rule: '[has(self.value), has(self.valueFrom)].exists_one(x, x)'
+                type: object
```

Comment:
these validations seem to be problematic when applying them in a cluster with `make install`:

```
The CustomResourceDefinition "packages.packages.glasskube.dev" is invalid: 
* spec.validation.openAPIV3Schema.properties[spec].properties[values].additionalProperties.properties[valueFrom].x-kubernetes-validations[0].rule: Forbidden: estimated rule cost exceeds budget by factor of 2.8x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared)
* spec.validation.openAPIV3Schema.properties[spec].properties[values].additionalProperties.x-kubernetes-validations[0].rule: Forbidden: estimated rule cost exceeds budget by factor of 2.3x (try simplifying the rule, or adding maxItems, maxProperties, and maxLength where arrays, maps, and strings are declared)
```

Reply:
Good catch! I found a different solution that works for these specific cases.